### DISCUSSION TITLE
Best way to handle dollar-formatted values in table?

### DISCUSSION THREAD

**Discussion by bryanwhiting** [h2oai/wave#1786] [Q&A]
If I convert my pandas dataframe to a string then the sorting doesn't work (because it does alphabetical sorting). Is there a dollar-format for the table column?

Alternatively, i could have white-text zero-padding the numbers and treat the column as html.

**Comment by vopani** [comment]
It might be a good feature to add support for currencies. It's common to have such data and I don't think there is a way to handle it today.

@mturoci How about adding a `ui.currency_cell_type(symbol='$')` that can be used to define the currency symbol with values continuing to be `data_type='number'` in `ui.table_column` and the UI can display the numbers along with the currency (kind of like how it is done in Microsoft Excel or any other way)?

<img width="85" alt="Screenshot 2023-01-11 at 10 52 21 AM" src="https://user-images.githubusercontent.com/8954217/211724082-73b2e2ff-a548-42f6-8746-d1e170312d47.png">

**Reply by mturoci** [reply]
One way to do this is through [serverside table](https://wave.h2o.ai/docs/examples/table-pagination/) which gives you control over every aspect of the table, sort included. The downside to this that with great power comes great responsibility (Uncle Ben) which means you will need to do all the heavy-lifting yourself.

I agree that we should make formatting a new feature within table col, but I would take it even further and add support for [generic formatting](https://wave.h2o.ai/docs/expressions/) instead. Either via a new cell_type as you suggested or a new col attribute for column in general, need to think it through.

**Reply by mturoci** [reply]
Tracked in https://github.com/h2oai/wave/issues/1791
